# Colorectal Cancer-Metagenomics
Whole Genome and Ampilicon Sequencing Study of Metagenomes of Colorectal Cancer Patients


# Background
- Colorectal cancer is the fourth most common cancer in men and the third most common cancer in women worldwide,
and significant international variations in the distribution of colorectal cancer have been observed.

- Once thought to be associated with developed nations whose populations were more susceptible to risk factors
(smoking, obesity, physical inactivity, alcohol etc), an increasing number of cases is recently being reported 
in developing nations, where it is currently being understudied.

- Signs and symptoms include worsening constipation, blood in stool, weight loss, nausea and vommiting. 

- However, around 50% of individuals with colorectal cancer do not report any symptom.

- This calls for a more thorough screening and detection procedure to complement current diagnostic procedures.

- Therapeutic interventions for colorectal cancer include surgery, chemotherapy, immunotherapy, radiation 
therapy and palliative care.


# Significance of the Study
- Without even showing symptoms, molecular markers have great precision and accuracy in detecting infections 
(including colorectal cancer) early stage, and can be relied on to aid in effective diagnosing of colorectal
cancers even in patients who may appear not to have them.


# Aims and Objectives
- Molecular identification of colorectal cancer will foster more effective detection and aid in more targetted 
treatment before complication.

- There are also a number of side effects associated with current interventions. Thus, early onset detection will maximize 
effective prognosis and minimize side-effects linked to diagnosis

- The aim of this study is therefore to identify genomic markers that can be used to screen for and identify early onset
colorectal cancer before complicating to forms difficult to treat. 

- We seek to achieve this by analyzing the 16S rRNA amplicon sequences and whole-genome sequences of colorectal cancer 
patients.


# Methodology
- Fecal samples of Colorectal Cancer Microbiome obtained from patients

- Marker gene sequencing (v6 hypervariable region of 16sRNA ampilicon sequencing)

- Taxonomic profiling with DADA2

- Diversity Analysis with alphadiversity

- Abundance and Functional Prediction with PICRUSt

## DADA2 Workflow
- [ ] Data Importing
- [ ] Quality Inspection
- [ ] Filtereing and Trimming
- [ ] Error Rate Estimation
- [ ] Dereplication
- [ ] Chimeras Removal
- [ ] Denoising and ASV Removal
- [ ] Taxonomic Classification

![Image of Yaktocat](https://slidetodoc.com/presentation_image_h/287504176fd61fbf9140bef6852f2963/image-24.jpg)

## PICRUSt Workflow
- [x] Predict 16s Copy Number abundance
- [x] Create the metagenome functional predictions 
- [x] Analyse predicted metagenomes

# References


